packages/suite-desktop/src-electron/modules/tor.ts

```diff
@@ -38,55 +38,81 @@ const init: Module = async ({ mainWindow, store, interceptor }) => {
         });
     };
 
+    const getProxySettings = (shouldEnableTor: boolean) =>
+        shouldEnableTor
+            ? {
+                  proxy: `socks://${address}`,
+                  useOnionLinks: true,
+              }
+            : { proxy: '', useOnionLinks: false };
+
     const setupTor = async (settings: TorSettings) => {
-        const shouldRunBundledTor = settings.running;
-        if (settings.running !== (await tor.status()).process) {
-            if (shouldRunBundledTor === true) {
-                try {
-                    await tor.start();
-                } catch (error) {
-                    logger.error('tor', `Failed to start: ${error.message}`);
-                    captureException(error);
-                }
-            } else {
-                try {
-                    await tor.stop();
-                } catch (error) {
-                    logger.error('tor', `Failed to stop: ${error.message}`);
-                    captureException(error);
-                }
-            }
+        const shouldEnableTor = settings.running;
+        const isTorRunning = (await tor.status()).process;
+
+        if (shouldEnableTor === isTorRunning) {
+            return;
         }
 
-        // Start (or stop) routing all communication through tor.
-        if (settings.running) {
+        if (shouldEnableTor === true) {
+            await tor.start();
+
             setProxy(`socks5://${host}:${port}`);
         } else {
+            await tor.stop();
             setProxy('');
         }
 
+        persistSettings(settings);
+
         // Notify the renderer.
         mainWindow.webContents.send('tor/status', settings.running);
```

Comment:
On toggle tor, tor status change is emitted twice in renderer layer.
<img width="600" alt="Screenshot 2022-06-23 at 8 51 51" src="https://user-images.githubusercontent.com/33235762/175234240-f17e2e62-d86e-427e-a248-f78a69c5c932.png">

Which results into this ugly bug:
Immediately when the tor is started, you click on the switch (because it is not disabled anymore), and it disables Tor but then another tor status comes and enables it again. And then it is disabled again.


https://user-images.githubusercontent.com/33235762/175234549-58e87a16-89ba-48d3-8835-36eaa6230285.mov

Reply:
Hopefully resolved by https://github.com/trezor/trezor-suite/pull/5651/commits/9fb0247f6482d184e1874323c46bddc4ba1e87c7